src/App.tsx

```diff
@@ -14,11 +15,107 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  list: string[]
+  sortBy: string
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    list: [],
+    sortBy: '',
+  };
+
+  showList = () => {
+    this.setState({ list: [...goodsFromServer] });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      list: state.list.reverse(),
+      sortBy: '',
+    }));
+  };
+
+  sortByAZ = () => {
+    this.setState({ sortBy: 'AZ' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({ list: [...goodsFromServer], sortBy: '' });
+  };
+
+  render() {
+    const { list = [''], sortBy } = this.state;
```

Comment:
```suggestion
    const { list, sortBy } = this.state;
```
Зачем там дефолтное значение, если у тебя изначально установлен массив как пустой в state